lib/index.ts

```diff
@@ -276,10 +276,14 @@ export default class Backburner {
       }
     }
 
-    if (length === 1) {
-      return method();
-    } else if (length === 2) {
-      return method.call(target);
+    let onError = getOnError(this.options);
+
+    if (onError) {
+      try {
```

Comment:
this PR should be merged anyway don't you think :P, because `try...catch` wrapping already exist in `.run` and not having `try...catch` in `.join` makes things inconsistent ? we can solve perf issues with later PRs